service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseManagementOrderService.java

```diff
@@ -39,7 +77,47 @@ public CaseManagementOrder addAction(CaseManagementOrder order, OrderAction orde
         return data;
     }
 
+    public CaseManagementOrder addHearingDetailsToCMO(DynamicList list, CaseManagementOrder order) {
+        CaseManagementOrder.CaseManagementOrderBuilder builder = order.toBuilder();
+
+        if (list != null) {
+            HearingDateDynamicElement hearingDateDynamicElement = hearingBookingService.getHearingDynamicElement(list);
+
+            builder
+                .action(order.getAction().toBuilder()
+                    .nextHearingId(hearingDateDynamicElement.getId())
+                    .nextHearingDate(hearingDateDynamicElement.getDate())
+                    .build())
+                .build();
+        }
+
+        return builder.build();
+    }
+
+    public String createNextHearingDateLabel(CaseManagementOrder caseManagementOrder,
+                                             List<Element<HearingBooking>> hearingBookings) {
+        String nextHearingLabel = "";
+
+        if (caseManagementOrder != null && caseManagementOrder.getAction() != null) {
+            UUID nextHearingId = caseManagementOrder.getAction().getNextHearingId();
+
+            HearingBooking hearingBooking =
```

Comment:
Can we remove this variable? It's only used in single line below.

Reply:
Or alternatively the whole method can be replaced with:

`        return Optional.ofNullable(caseManagementOrder)
            .map(CaseManagementOrder::getAction)
            .map(OrderAction::getNextHearingId)
            .map(id -> hearingBookingService.getHearingBookingByUUID(hearingBookings, id))
            .map(HearingBooking::getStartDate)
            .map(this::formatHearingBookingLabel)
            .orElse("");
`

Reply:
If you like the FUNCTIONAL 😈